addons/binding/org.openhab.binding.tankerkoenig/src/main/java/org/openhab/binding/tankerkoenig/handler/BridgeHandler.java

```diff
@@ -0,0 +1,216 @@
+        this.setRefreshInterval(((BigDecimal) config.get(TankerkoenigBindingConstants.CONFIG_REFRESH)).intValue());
+        this.setSetupMode((boolean) config.get(TankerkoenigBindingConstants.CONFIG_SETUP_MODE));
+        updateStatus(ThingStatus.ONLINE);
+
+        int pollingPeriod = this.getRefreshInterval();
+        pollingJob = scheduler.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+                logger.debug("Try to refresh data");
+                // Just update data if setupMode is false(off)
+                if (!isSetupMode()) {
+                    UpdateTankstellenData();
+
+                    UpdateTankstellenThings();
+
+                }
+            }
+        }, 15, pollingPeriod * 60, TimeUnit.SECONDS);
+
+        logger.debug("Refresh job scheduled to run every {} min. for '{}'", pollingPeriod, getThing().getUID());
+
+    }
+
+    /***
+     *
+     * @param tankstelle
+     * @return
+     */
+    public boolean RegisterTankstelleThing(Thing tankstelle) {
+
+        if (this.tankstellenThingList.size() == 10) {
+            return false;
+        }
+
+        logger.info("Tankstelle " + tankstelle.getUID().toString() + "was registered to config "
+                + this.getThing().getUID().toString());
+
+        this.tankstellenThingList.add(tankstelle);
+
+        return true;
+
+    }
+
+    public void UnregisterTankstelleThing(Thing tankstelle) {
+        logger.info("Tankstelle " + tankstelle.getUID().toString() + "was unregistered from config "
+                + this.getThing().getUID().toString());
+        this.tankstellenThingList.remove(tankstelle);
+
+    }
+
+    /***
+     * Updates the data from tankerkoenig api (no update on things)
+     */
+    public void UpdateTankstellenData() {
+        // Get data
+        String locationIDsString = GenerateLocationIDsString();
+
+        if (locationIDsString.length() < 1) {
+            logger.info("No tankstellen id's found. Nothing to update");
+            return;
+        }
+
+        TankerkoenigService service = new TankerkoenigService();
+        TankerkoenigListResult result = service.GetTankstellenListData(this.getApiKey(), locationIDsString);
+
+        this.setTankerkoenigListResult(result);
+
+        this.tankstellenList.clear();
+        for (LittleStation station : result.getPrices().getStations()) {
+            this.tankstellenList.put(station.getID(), station);
+        }
+
+    }
+
+    /***
+     * Updates all registered Tankstellen with new data
+     */
+    public void UpdateTankstellenThings() {
+
+        for (Thing thing : tankstellenThingList) {
+            TankerkoenigHandler tkh = (TankerkoenigHandler) thing.getHandler();
+            LittleStation s = this.tankstellenList.get(tkh.getLocationID());
+            if (s == null) {
+                logger.info("Could not find tankstelle with id " + tkh.getLocationID());
+                continue;
```

Comment:
Using continue here is not the best Java style.  Better use if/else in this case